shm/inc/map.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <vector>
+
+#include "island.hpp"
```

Comment:
```suggestion
#include "island.hpp"

#include <vector>
```
The other files have include order as above. First coresponding header, then local header, then system header. I think it's good idea to stick to one convention. But it's up to you.

Reply:
It's the other way around.
See https://coders.school/edu/courses/powtorka-1-qa/lesson/kolejnosc-include/.

I have my VSC set in a way, that would revert the order of headers automatically if i applied your suggestion

Reply:
(it's enabled by default by the chromium style)